python/cairo-core/src/cairo_core/maths.cairo

```diff
@@ -536,3 +536,57 @@ func felt252_bit_length{range_check_ptr}(value: felt) -> felt {
 
     return bit_length;
 }
+
+func felt252_to_bits{range_check_ptr}(limb: felt, bits_ptr: felt*, current_len: felt) -> felt {
+    alloc_locals;
+    if (limb == 0) {
+        return current_len;
+    }
+    with_attr error_message("felt252_to_bits: limb must be < 252") {
+        assert [range_check_ptr] = limb;
+        assert [range_check_ptr + 1] = 251 - limb;
+        let range_check_ptr = range_check_ptr + 2;
+    }
+
+    let output = &bits_ptr[0];
+    %{ felt252_to_bits %}
+
+    tempvar range_check_ptr = range_check_ptr;
+    tempvar current_len = current_len;
+    tempvar i = 0;
+    tempvar acc = 0;
+
+    loop:
+    let range_check_ptr = [ap - 3];
+    let current_len = [ap - 2];
+    let limb = [ap - 1];
+
+    let is_done = is_zero(limb);
+    jmp end if is_done != 0;
+
+    // Get the current bit using modulo 2
+    let (_, bit) = unsigned_div_rem(limb, 2);
+    assert bits_ptr[current_len] = bit;
```

Comment:
this is the right verification, however, if we do a division of `value/2` to get the current bit, this defeats the purpose of this function - as we want to rely on multiplications only, that are much cheaper than divisions.

This is why we have an accumulator and verify that sum(bit[i] << i) = original_value